core/src/main/java/org/opensearch/sql/expression/datetime/DateTimeFunction.java

```diff
@@ -1364,6 +1373,38 @@ private ExprValue exprQuarter(ExprValue date) {
     return new ExprIntegerValue((month / 3) + ((month % 3) == 0 ? 0 : 1));
   }
 
+  private double roundToZero(double val) {
+    if (val < 0) {
+      return Math.ceil(val);
+    }
+    return Math.floor(val);
+  }
+
+  private ExprValue exprSecToTime(ExprValue totalSeconds) {
+
+    try {
+      int hours = totalSeconds.integerValue() / 3600;
+      int minutes = (totalSeconds.integerValue() % 3600) / 60;
+      int seconds = totalSeconds.integerValue() % 60;
+      String fraction = String.valueOf(
+          totalSeconds.doubleValue() - roundToZero(totalSeconds.doubleValue()));
+
+      if (fraction == "0") {
+        return new ExprTimeValue(String.format("%02d:%02d:%02d", hours, minutes, seconds));
+      }
+
+      return new ExprTimeValue(String.format("%02d:%02d:%02d.%s",
+          hours,
+          minutes,
+          seconds,
+          fraction.substring(2)));
+
+    } catch (Exception e) {
```

Comment:
Please don't do this.  This will catch any error and hide it. 
Whenever using try...catch, be as specific as possible with the Exception and the line numbers (so if you're trying to catch parsing failures, include the try...catch around your parsing statements only.

Reply:
Fixed in https://github.com/Bit-Quill/opensearch-project-sql/pull/219/commits/9ae0057b2cc009e94c396c8713bc68d57020b1db